api/src/lib.rs

```diff
@@ -584,30 +610,49 @@ impl Client {
             .json::<Response<EmptySuccess, ErrorT>>()
             .chain_err(|| ErrorKind::BadJsonResponse)?
             .into_result(status)
-            .map(|_| ())
+            .map_or_else(
+                // Ignore 404 not found if the request had to be re-tried - assume the target
+                // object was deleted on a previous incomplete request.
+                |e| {
+                    if attempts.get() > 1 && status == 404 {
+                        Ok(())
+                    } else {
+                        Err(e)
+                    }
+                },
+                |_| Ok(()),
+            )
     }
 
     fn post<LocationT, RequestT, SuccessT, ErrorT>(
         &self,
         url: LocationT,
         request: RequestT,
+        retry: bool,
```

Comment:
This is a nit, but my preference would be to have `enum Retry { No, Yes }` and take that as argument, making calls to `post` more readable, `post(endpoint, request, Retry::Yes)` over `post(endpoint, request, true)`.

Reply:
Very happy to use enums instead of bools if that's our preferred style. Increased verbosity is why I didn't opt for it initially, though completely agree it can aid interpretation at the call site.